### PR TITLE
add contributing doc

### PR DESCRIPTION
**Pull request by tpepper**
Info from README.md and MAINTAINERS.md is updated and merged into a new
CONTRIBUTING.md.

Signed-off-by: Tim Pepper <tpepper@vmware.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+## Quality Controls
+
+This needs enhancing.  ``scripts/`` and ``travis.sh`` include some
+initial plumbing.  [Travis CI](https://travis-ci.org/) will run on
+your github fork and its branches and also on your PR's to ``u-root``.
+
+## Discussion
+
+``u-root`` is on Slack as "u-root.slack.com".
```
Comment (ggiamarchi) [inline CONTRIBUTING.md]: Sign up on Slack is restricted to a few domains: `@google.com`, `@princeton.edu`, `@mit.edu` and `@kochris.com`. As an open source project, I guess it should be open for all.

Comment (hugelgupf) [reply CONTRIBUTING.md]: Slack doesn't let you do that easily. We're going to have to set up some kind of other service to let people register for our slack. (I tried adding @gmail.com to that list of domains, and slack explicitly rejected that domain...)

**Comment by rminnich** [comment]
we should ask slack what it takes to make that work.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+We accept github pull requests.
+
+Fork the project on github, work in your fork and in branches, push
+these to your github fork, and when ready, do a github pull requests
+against https://github.com/u-root/u-root.
+
+## Code Reviews
+
+Look at the area of code you're modifying, it's history, and consider
```
Comment (rminnich) [inline CONTRIBUTING.md]: it's -> its

File: CONTRIBUTING.md (hunk 1 above)
Comment (rminnich) [inline CONTRIBUTING.md]: "until we figure out how to allow anyone to sign up, please submit an issue asking to be added to slack. Sorry for the inconvenience"

File: MAINTAINERS.md [hunk 3]
```diff
@@ -1,46 +1,40 @@
...
-    import (
-      "regexp"
-      "time"
+# Style Guide
 
-      dhcp "github.com/krolaw/dhcp4"
-    )
-    ```
+In [CONTRIBUTING.md](CONTRIBUTING.md) we say `gofmt` and `golint` are law,
+but that's not enforced (yet) in automatino.
```
Comment (rminnich) [inline MAINTAINERS.md]: automation

**Comment by hugelgupf** [comment]
Ron, I assume that it's intentional. I assume that it's a feature of their paid tier to let guests come into the chatroom and/or to let randos register.

File: CONTRIBUTING.md (hunk 2 above)
Comment (tpepper) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 1 above)
Comment (tpepper) [reply CONTRIBUTING.md]: added

File: MAINTAINERS.md (hunk 3 above)
Comment (tpepper) [reply MAINTAINERS.md]: fixed

**Pull request merged**